js/packages/openinference-instrumentation-openai/src/instrumentation.ts

```diff
@@ -334,17 +337,64 @@ function getLLMInputMessagesAttributes(
   body: ChatCompletionCreateParamsBase,
 ): Attributes {
   return body.messages.reduce((acc, message, index) => {
+    const messageAttributes = getChatCompletionInputMessageAttributes(message);
     const index_prefix = `${SemanticConventions.LLM_INPUT_MESSAGES}.${index}`;
-    acc[`${index_prefix}.${SemanticConventions.MESSAGE_CONTENT}`] = String(
-      message.content,
-    );
-    acc[`${index_prefix}.${SemanticConventions.MESSAGE_ROLE}`] = String(
-      message.role,
-    );
+    // Flatten the attributes on the index prefix
+    for (const [key, value] of Object.entries(messageAttributes)) {
+      acc[`${index_prefix}.${key}`] = value;
+    }
     return acc;
   }, {} as Attributes);
 }
 
+function getChatCompletionInputMessageAttributes(
+  message: ChatCompletionMessageParam,
+): Attributes {
+  const role = message.role;
+  const attributes: Attributes = {
+    [SemanticConventions.MESSAGE_ROLE]: role,
+  };
+  switch (role) {
```

Comment:
`content` should be outside the `switch`. As written, it's not being captured for `assistant`.
```suggestion
  };
  if (message.content)
      attributes[SemanticConventions.MESSAGE_CONTENT] = message.content;
  switch (role) {
```

Reply:
Yeah was trying to make the switch blocks more self contained but I'll hoist this